pkg/arvo/sys/hoon.hoon

```diff
@@ -1583,19 +1581,19 @@
     =.  n.a  +.d
     =+  e=$(a l.a, b -.d)
     =+  f=$(a r.a, b -.e)
-    [-.f [n.a +.e +.f]]
+    [-.f a(l +.e, r +.f)]
   ::
   ++  run                                               ::  apply gate to values
     |*  b/gate
     |-
     ?~  a  a
-    [n=[p=p.n.a q=(b q.n.a)] l=$(a l.a) r=$(a r.a)]
+    a(n n.a(q (b q.n.a)), l $(a l.a), r $(a r.a))
```

Comment:
`a(q.n (b q.n.a), l $(a l.a), r $(a r.a))`

This line replaces so much of the noun that it might actually be better as an autocons:
`[[p.n.a (b q.n.a)] $(a l.a) $(a r.a)]`

Reply:
In that case I'd suggest the autocons version with faces this is originally replacing

Reply:
@ohAitch I think you're right, especially since this is wet.  The type system might actually need the faces.

Reply:
"Need" is a social construct; but it's certainly the case that if you call the faceless `run` two times in a row, then the second one won't be able to pull `q.n.a`, because the first didn't give it a `q` or an `n` 😉

(Unless you cast `a` back to a map in the middle, which we certainly find ourselves needing to do on occasion, though I think usually for more subtle reasons than this one)

Reply:
I suppose actually, I completely agree with framing things like iron core nesting as the type system "demanding" or "compelling" specific behavior; but in the particular case of putting faces on data structures in wet gates(or dry gates samples, or other casts), the type system is just doing exactly what you told it to.

Reply:
Sure, I mostly meant that in practice we're better off putting faces on the product so that passing the result into other wet code will be less likely to run into problems.  I think that's pretty similar to what you're saying.